File: Application/SmartWeather.Api/Controllers/Component/Dtos/ComponentResponse.cs
using SmartWeather.Api.Controllers.MeasurePoint.Dtos;
using SmartWeather.Entities.Component;
using SmartWeather.Entities.MeasurePoint;

namespace SmartWeather.Api.Controllers.Component.Dtos;

public class ComponentResponse
{
    public int Id { get; set; }
    public int GpioPin { get; set; }
    public string Name { get; set; } = null!;
    public string Color { get; set; } = null!;
    public ComponentType Type { get; set; }
    public int StationId { get; set; }
    public List<MeasurePointResponse>? MeasurePoints { get; set; }
}

File: Presentation/SmartWeather.WebClient/src/services/component/dtos/component_response.ts
import { MeasurePointResponse } from "@/services/measurepoint/dtos/measurepoint_response"

export interface ComponentResponse {
    id: number
    gpioPin: number
    name: string
    color: string
    type: ComponentType
    stationId: number
    measurePoints: MeasurePointResponse[] | undefined;
}

export enum ComponentType {
    Unknown,
    Sensor,
    Actuator
}